orchestrator/careplancontributor/applaunch/zorgplatform/handle_create_hcp_rst.go

```diff
@@ -26,6 +26,35 @@ type SecureTokenService interface {
 
 var _ SecureTokenService = &Service{}
 
+func (s *Service) RequestApplicationToken(launchContext LaunchContext) (string, error) {
+
+	assertion, err := s.createSAMLAssertionForApplication(&launchContext)
+	if err != nil {
+		return "Failed to create SAML assertion", err
```

Comment:
just wrap the error?

```suggestion
		return fmt.Errorf("failed to create SAML assertion: %w", err)
```

(also applies to the ones below)